include/tdi/common/tdi_table_data.hpp

```diff
@@ -328,6 +328,14 @@ class TableData {
    */
   const tdi_id_t &actionIdGet() const { return action_id_; };
 
+  /**
+   * @brief Set actionId
+   *
+   * @param[in] tdi_id_t action ID
+   *
+   */
+  void actionIdSet(tdi_id_t action_id) { this->action_id_ = action_id; }
```

Comment:
Can't we use the reset function for this though ?

Reply:
Yeah, reset can be used. reverted this change.